File: backend/GraficheVeneziane/src/main/java/graficheveneziane/graficheveneziane/model/Prodotto.java
package graficheveneziane.graficheveneziane.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import java.util.Set;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Prodotto {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id_prodotto;

    @Column
    private String nome_prodotto;

    @Column
    private String descrizione_prodotto;

    @JsonIgnore
    @OneToMany(mappedBy = "prodotto")
    private Set<OrdineProdotto> ordini;


    @ManyToOne()
    @JoinColumn(name = "id_fornitore")
    private Fornitore fornitore;

    @Column
    private int prezzo;
}

File: graficheFE/src/app/model/Prodotto.ts
import { Fornitore } from "./Fornitore";

export class Prodotto {
  id_prodotto: number;
  nome_prodotto: string;
  descrizione_prodotto: string;
  fornitore: Fornitore;
  prezzo: number;

  constructor(id_prodotto: number, nome_prodotto: string, descrizione_prodotto: string,  fornitore: Fornitore, prezzo: number) {
    this.id_prodotto = id_prodotto;
    this.nome_prodotto = nome_prodotto;
    this.descrizione_prodotto = descrizione_prodotto;
    this.fornitore = fornitore;
    this.prezzo = prezzo;
  }
}
